endpoint/endpoint.go

```diff
@@ -0,0 +1,24 @@
+package endpoint
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Endpoint knows how to connect with a Transport or a Transfer
+type Endpoint interface {
+	// NamespacedName returns a ns name to identify this endpoint
+	NamespacedName() types.NamespacedName
+	// Hostname returns a hostname for the endpoint
+	Hostname() string
+	// BackendPort returns a port which can be used by the application behind this endpoint
+	BackendPort() int32
+	// IngressPort is a port which is used by the clients to connect to the endpoint
+	IngressPort() int32
+	// IsHealthy returns whether or not all Kube resources used by endpoint are healthy
+	IsHealthy(ctx context.Context, c client.Client) (bool, error)
+	// MarkForCleanup
+	MarkForCleanup(ctx context.Context, c client.Client, key, value string) error
```

Comment:
I assume they will pass in and remember the k/v that gets applied to the underlying objects, but how will they do the actual cleanup? Are they expected to do the `DeleteAllOf` on the types that are returned by a call to `APIsToWatch`?

Reply:
refactored, thanks!

Reply:
> refactored, thanks!

^ that was in response to a different comment.

But yes, I imagine APIsToWatch feeding the cleanup APIs as well. Open to renaming the function as appropriate.